<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="mode" default="$(env MOVEL_MODE)" doc="mode: [deployment, development]"/>
  <arg name="camera" default="camera"/>
  <remap from="cmd_vel_mux/autonomous" to="cmd_vel_mux/capped" />

  <node pkg="pallet_detection" name="apriltag_detection" type="apriltag_detection_node" output="screen">
    <rosparam file="$(find pallet_detection)/config/apriltag_detection.yaml"/>
    <rosparam if="$(eval mode == 'deployment')" file="$(env HOME)/.config/movel/config/pallet_detection/config/apriltag_detection.yaml"/>
    <param name="use_move_base" value="false"/>
  </node>

  <node pkg="pallet_detection" name="pallet_docking" type="pallet_docking" output="screen">
    <rosparam file="$(find pallet_detection)/config/pallet_docking.yaml"/>
    <rosparam if="$(eval mode == 'deployment')" file="$(env HOME)/.config/movel/config/pallet_detection/config/pallet_docking.yaml"/>
    <param name="use_move_base" value="false"/>
    <param name="retry" value="false"/>
    <param name="undock" value="false"/>
  </node>
	
  <include file="$(find planner_adjuster)/launch/planner_adjuster.launch"/>

  <!-- April tag publisher -->
  <arg name="april_node_namespace" default="apriltag_ros_continuous_node" />
  <rosparam command="load" file="$(find pallet_detection)/config/april_settings.yaml" ns="$(arg april_node_namespace)" />
  <rosparam command="load" file="$(find pallet_detection)/config/april_tags.yaml" ns="$(arg april_node_namespace)" />
  <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="$(arg april_node_namespace)" clear_params="true" output="screen" >
    <remap from="image_rect" to="/$(arg camera)/color/image_raw" />
    <remap from="camera_info" to="/$(arg camera)/color/camera_info" />
    <param name="camera_frame" type="str" value="$(arg camera)_color_optical_frame" />
    <param name="publish_tag_detections_image" type="bool" value="false" /> 
  </node>
</launch>
